pennylane/templates/broadcast.py

```diff
@@ -538,33 +540,22 @@ def circuit(pars):
                 "the ring pattern with 2 wires is an exception and only applies one unitary"
             )
 
-        if shape[0] != n_parameters[pattern]:
+        if shape[0] != pattern_to_num_params[pattern]:
             raise ValueError(
                 "'parameters' must contain entries for {} unitaries; got {} entries".format(
-                    n_parameters[pattern], shape[0]
+                    pattern_to_num_params[pattern], shape[0]
                 )
             )
 
-        # repackage for consistent unpacking
+        # expand dimension so that parameter sets for each unitary can be unpacked
         if len(shape) == 1:
-            parameters = [[p] for p in parameters]
-    else:
-        parameters = [[] for _ in range(n_parameters[pattern])]
-
-    #########
-
-    # define wire sequences for patterns
-    wire_sequence = {
-        "single": [wires[i] for i in range(len(wires))],
-        "double": [wires.subset([i, i + 1]) for i in range(0, len(wires) - 1, 2)],
-        "double_odd": [wires.subset([i, i + 1]) for i in range(1, len(wires) - 1, 2)],
-        "chain": [wires.subset([i, i + 1]) for i in range(len(wires) - 1)],
-        "ring": wires_ring(wires),
-        "pyramid": wires_pyramid(wires),
-        "all_to_all": wires_all_to_all(wires),
-        "custom": custom_pattern,
-    }
+            parameters = qml.tape.interfaces.functions.WrapperFunctions.expand_dims(parameters, 1)
 
-    # broadcast the unitary
-    for wires, pars in zip(wire_sequence[pattern], parameters):
-        unitary(*pars, wires=wires, **kwargs)
```

Comment:
Below version is cleaner, because it does not replace no parameters with empty parameters.